File: JDK11/opentraceability/src/main/java/opentraceability/models/events/eventdestination.java
package opentraceability.models.events;

import opentraceability.utility.CBVAttribute;
import opentraceability.utility.EnumUtil;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;

public class EventDestination {
    @OpenTraceabilityJsonAttribute("type")
    @OpenTraceabilityAttribute("", "@type")
    public URI type;

    @OpenTraceabilityJsonAttribute("destination")
    @OpenTraceabilityAttribute("", "text()")
    public String value;

    public EventDestinationType getParsedType() {
        EventDestinationType type = EventDestinationType.Unknown;
        for (EventDestinationType e : EventDestinationType.values()) {
            for (CBVAttribute attribute : EnumUtil.GetEnumAttributes(e, CBVAttribute.class)) {
                if (attribute.value.toLowerCase().equals(type.toString().toLowerCase())) {
                    return e;
                }
            }
        }
        return type;
    }

    public void setParsedType(EventDestinationType value) {
        String t = EnumUtil.GetEnumAttributes(value, CBVAttribute.class).stream().filter(a -> a.value.startsWith("urn")).findFirst().get().value;
        if (!t.isBlank()) {
            this.type = URI.create(t);
        }
    }

}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/events/eventdestination.kt
package opentraceability.models.events

import opentraceability.utility.CBVAttribute
import opentraceability.utility.EnumUtil
import opentraceability.utility.attributes.*
import java.net.URI

class EventDestination {

    @OpenTraceabilityJsonAttribute("type")
    @OpenTraceabilityAttribute("","@type")
    var type: URI? = null

    @OpenTraceabilityJsonAttribute("destination")
    @OpenTraceabilityAttribute("","text()")
    var value: String? = null

    var ParsedType: EventDestinationType
        get() {
            var type = EventDestinationType.Unknown

            for (e in enumValues<EventDestinationType>()) {
                if (EnumUtil.GetEnumAttributes<EventDestinationType, CBVAttribute>(e)
                        .any { it.value.toLowerCase() == type?.toString()?.toLowerCase() }
                ) {
                    return e
                }
            }

            return type
        }
        set(value) {
            val t = EnumUtil.GetEnumAttributes<EventDestinationType, CBVAttribute>(value).firstOrNull { it.value.startsWith("urn") }?.value
            if (!t.isNullOrBlank()) {
                this.type = URI(t)
            }
        }

}
